src/Avalonia.Controls/Control.cs

```diff
@@ -237,6 +343,17 @@ protected override void OnAttachedToVisualTree(VisualTreeAttachmentEventArgs e)
             InvalidateMirrorTransform();
         }
 
+        /// <inheritdoc/>
+        public override void Render(DrawingContext context)
+        {
+            if (_isLoaded == false)
```

Comment:
Small concern here: If someone overrides Render and leaves out `base.Render(context)` this gets skipped so `Loaded` won't fire and `IsLoaded` will always be false.

Reply:
Yea, I agree it shouldn't be possible to break unrelated functionality quite so easily. I'll find where the Render() method is first called and try to invoke OnLoadedCore() from there instead.

Reply:
This is too tricky to change I think. Following the call stack down to where Render originates of course gets into the renders which only have access to the Visual. I would have to add a dependency to Control to call it from here which would break the architecture I think. So hopefully the solution mentioned above works instead.

This should be obsolete now so resolving.